nixos/modules/services/networking/go-neb.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.go-neb;
+
+  configFile = pkgs.writeText "config.yml" (builtins.toJSON cfg.config);
+in {
+  options.services.go-neb = {
+    enable = mkEnableOption "Extensible matrix bot written in Go";
+
+    bindAddress = mkOption {
+      type = types.str;
+      description = "Port (and optionally address) to listen on.";
+      default = ":4050";
+    };
+
+    baseUrl = mkOption {
+      type = types.str;
+      description = "Public-facing endpoint that can receive webhooks.";
+    };
+
+    config = mkOption {
+      type = types.attrs;
```

Comment:
Multiple definitions of attrs are apparently not merged properly. [According to @Infinisil, `types.uniq types.attr` might be appropriate](https://logs.nix.samueldr.com/nixos-dev/2020-06-02#3549491;) to prevent that merging.

Reply:
That mistake is probably all over nixpkgs right? But seems easy enough to fix here.

Reply:
In fact i don't see that used anywhere. Wild!